gencert-private.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/sh
+
+# generate configuration file
+
+openssl_conf=$(mktemp)
+
+cat > "${openssl_conf}" <<EOF
+oid_section=spid_oids
+
+[ req ]
+default_bits=2048
+default_md=sha256
+distinguished_name=dn
+encrypt_key=no
+prompt=no
+req_extensions=req_ext
+
+[ spid_oids ]
+agid=1.3.76.16
+spid-privatesector-SP=1.3.76.16.4.3.1
+spid-publicsector-SP=1.3.76.16.4.2.1
+uri=2.5.4.83
+
+[ dn ]
+commonName=${ENTITY_ID}
+countryName=IT
+localityName=${LOCALITY_NAME}
+stateOrProvinceName=${PROVINCIE}
+organizationIdentifier=VATID-${PIVA}
+organizationName=${ORGANIZATION_NAME}
+serialNumber=VATID-${PIVA}
+uri=${ENTITY_ID}
+emailAddress=${EMAIL}
+
+[ req_ext ]
+basicConstraints=CA:FALSE
+keyUsage=critical,digitalSignature,nonRepudiation
+certificatePolicies=@agid_policies,@spid_policies
+
+[ agid_policies ]
+policyIdentifier=agid
+userNotice=@agid_notice
+
+[ agid_notice ]
+explicitText="cert_SP_Privati"
+
+[ spid_policies ]
+policyIdentifier=spid-privatesector-SP
+userNotice=@spid_notice
+
+[ spid_notice ]
+explicitText="Service Provider SPID Privato"
+EOF
+
+cat <<EOF
+## --------------------------------------------------------------------------
+## Generated OpenSSL configuration
+## --------------------------------------------------------------------------
+$(cat ${openssl_conf})
+EOF
+
+## --------------------------------------------------------------------------
+## Generate certificate private key and signing request
+## --------------------------------------------------------------------------
+
+openssl req -config ${openssl_conf} -new -newkey rsa:2048 -nodes -keyout privkey.pem -out CSR.csr -extensions req_ext 2>/dev/null
```

Comment:
The switches `-newkey rsa:2048 -nodes` overwrite the settings `default_bits` and `encrypt_key` in `openssl_conf`. Remove them from the OpenSSL command.

For the sake of readability, I would suggest to rewrite the command as multi-line, like in the following

```
openssl req -config ${openssl_conf} -new \
   -keyout privkey.pem -out CSR.csr \
   -extensions req_ext 2>/dev/null
```

About out files (private key and CSR), please use variables as in the first comment.